tests/frameworks/tf_keras/test_tf_keras.py

```diff
@@ -0,0 +1,230 @@
+        verbose=0,
+    )
+
+    return model
+
+
+def evaluate(model: tf.keras.Model) -> dict:
+    # Load the model:
+    model_path = None
+    if not isinstance(model, tf.keras.Model):
+        model_path = model
+        model_handler = TFKerasModelHandler(model_path=model)
+        model_handler.load()
+        model = model_handler.model
+
+    # Load the MNIST dataset
+    print("\nLoading MNIST dataset...")
+    (_, _), (x_test, y_test) = tf.keras.datasets.mnist.load_data()
+
+    # Preprocess the data
+    x_test, y_test = preprocess_data(x=x_test, y=y_test)
+
+    print("\nApplying MLRun to the model...")
+    apply_mlrun(model=model, model_path=model_path, model_name="mnist_cpu_model")
+
+    # Evaluate the model on the test data
+    print("\nEvaluating the model on the test set...")
+    loss, accuracy = model.evaluate(x_test, y_test, verbose=0)
+    print(f"Test Loss: {loss:.4f}")
+    print(f"Test Accuracy: {accuracy:.4f}")
+
+    return {
+        "loss": loss,
+        "accuracy": accuracy,
+    }
+
+
+@pytest.mark.skipif(
+    is_keras_3(),
+    reason="Keras 3 test is freezing due to dead lock issue when running within PyTest.",
+)
+def test_training(rundb_mock):
+    """
+    Test the `apply_mlrun` function with a simple TensorFlow Keras model training.
+    """
+    test_directory = tempfile.TemporaryDirectory()
+
+    # Run training:
+    train_run = mlrun.new_function().run(
+        artifact_path=test_directory.name,
+        handler=train,
+        local=True,
+    )
+
+    # Print the outputs for manual validation:
+    print(json.dumps(train_run.outputs, indent=4))
+
+    # Get assertion parameters:
+    expected_artifacts = [
+        "accuracy_summary.html",
+        "learning_rate_values.html",
+        "loss_summary.html",
+        "model",
+        "training_accuracy.html",
+        "training_loss.html",
+        "validation_accuracy.html",
+        "validation_loss.html",
+    ]
+    expected_results = [
+        "learning_rate",
+        "training_accuracy",
+        "training_loss",
+        "validation_accuracy",
+        "validation_loss",
+    ]
+
+    # Validate artifacts:
+    for expected_artifact in expected_artifacts:
+        assert expected_artifact in train_run.status.artifact_uris
+    assert len(train_run.status.artifacts) == len(expected_artifacts)
+
+    # Validate results:
+    for expected_result in expected_results:
+        assert expected_result in train_run.status.results
+    assert (
+        len(train_run.status.results) == len(expected_results) + 1
+    )  # +1 for the returned model.
+
+    test_directory.cleanup()
+
+
+@pytest.mark.skipif(
+    is_keras_3(),
+    reason="Keras 3 test is freezing due to dead lock issue when running within PyTest.",
+)
+def test_evaluation(rundb_mock):
+    """
+    Test the `apply_mlrun` function with a simple TensorFlow Keras model training to evaluation flow.
+    """
+    test_directory = tempfile.TemporaryDirectory()
```

Comment:
ensures cleanup if test fails
```suggestion
    with tempfile.TemporaryDirectory() as tmpdir:
```